pkg/solana/client/client.go

```diff
@@ -151,6 +153,43 @@ func (c *Client) GetFeeForMessage(msg string) (uint64, error) {
 	return *res.Value, nil
 }
 
+// https://docs.solana.com/developing/clients/jsonrpc-api#getsignaturestatuses
+func (c *Client) SignatureStatuses(ctx context.Context, sigs []solana.Signature) ([]*rpc.SignatureStatusesResult, error) {
+	ctx, cancel := context.WithTimeout(ctx, c.contextDuration)
+	defer cancel()
+
+	// searchTransactionHistory = false
+	res, err := c.rpc.GetSignatureStatuses(ctx, false, sigs...)
+	if err != nil {
+		return nil, errors.Wrap(err, "error in GetSignatureStatuses")
+	}
+
+	if res == nil || res.Value == nil {
+		return nil, errors.New("nil pointer in GetSignatureStatuses")
+	}
+	return res.Value, nil
+}
+
+// https://docs.solana.com/developing/clients/jsonrpc-api#simulatetransaction
+func (c *Client) SimulateTx(ctx context.Context, tx *solana.Transaction) (*rpc.SimulateTransactionResult, error) {
+	ctx, cancel := context.WithTimeout(ctx, c.contextDuration)
+	defer cancel()
+
+	res, err := c.rpc.SimulateTransactionWithOpts(ctx, tx, &rpc.SimulateTransactionOpts{
```

Comment:
Consider adding an optional input for `opts` so the user could use them, or if `nil` the fn uses these defaults.

Reply:
added!